#!/bin/bash
# -P eventlet -> threads
# eventlet method will result in an error -> django.db.utils.DatabaseError: DatabaseWrapper objects created in a thread can only be used in that same thread. The object with alias 'default' was created in thread id 140032074089736 and this is thread id 140031737889904.
# this problem solution is not use eventlet
# refer to -> https://github.com/celery/celery/issues/5924
# refer to -> https://github.com/intelowlproject/IntelOwl/issues/379
celery -A apps.backend worker -Q service_schedule,service_schedule_priority -n schedule_worker@%h -P threads -c 100 --max-tasks-per-child=50 -O fair --time-limit=1800
